kong/tools/utils.lua

```diff
@@ -942,9 +942,12 @@ _M.check_hostname = function(address)
   end
 
   -- Reject prefix/trailing dashes and dots in each segment
-  -- note: punycode allowes prefixed dash, if the characters before the dash are escaped
-  for _, segment in ipairs(split(name, ".")) do
-    if segment == "" or segment:match("-$") or segment:match("^%.") or segment:match("%.$") then
+  -- notes:
+  --   - punycode allowes prefixed dash, if the characters before the dash are escaped
```

Comment:
Let's fix the typos while we're here
```suggestion
  --   - punycode allows prefixed dash, if the characters before the dash are escaped
```